### PR TITLE
Issue/3232 viewbinding product detail p2

### PR DESCRIPTION
**Pull request by nbradbury**
This is the 2nd PR to convert product detail to view binding, and it tackles the category screens. 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ImageViewerFragment.kt [hunk 1]
```diff
@@ -41,23 +40,30 @@ class ImageViewerFragment : androidx.fragment.app.Fragment(), RequestListener<Dr
...
         loadImage()
-        photoView.setOnPhotoTapListener { _, _, _ ->
+        binding.photoView.setOnPhotoTapListener { _, _, _ ->
             imageListener?.onImageTapped()
         }
     }
 
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ImageViewerFragment.kt]: I don't believe I caught this before, but I believe `_binding = null` should be called _before_  `super.onDestroyView()`.  Same with any other instances in code. [It's been a rule of thumb I've used after reading "but it's a general rule: during any kind of initialization, let the super class do their work first; during any kind of finalization, you do your work first.".](https://groups.google.com/g/android-developers/c/ndsrBsIchFc/m/z-3xVTT0EjQJ). Sorry if I missed these before. I'm not sure it always makes a big difference, but it seems to be safer in practice and less prone to bugs.

**Review by AmandaRiu — COMMENTED** [review]
@nbradbury Changes look good and everything works fine. I left a code comment for your review. There may be a reason you are doing it this way, but I just wanted to make sure.

**Comment by AmandaRiu** [comment]
@nbradbury  Also, there is a merge conflict.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ImageViewerFragment.kt (hunk 1 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ImageViewerFragment.kt]: > I don't believe I caught this before, but I believe `_binding = null` should be called _before_ `super.onDestroyView()`

I followed [Google's example code here](https://developer.android.com/topic/libraries/view-binding), which nulls the binding after `super.onDestroyView()` is called. I can change this everywhere if you feel strongly about it, but I don't think it's really necessary.

> Also, there is a merge conflict.

Resolved.

Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ImageViewerFragment.kt]: That's okay, if you're good with it, then I am as well.

**Review by AmandaRiu — APPROVED** [review]
:shipit:

**Pull request merged**